plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -317,9 +328,20 @@ else if (javaType == Slice.class) {
     public CallableStatement callProcedure(JdbcClient client, ConnectorSession session, Connection connection, ProcedureQuery procedureQuery)
             throws SQLException
     {
+        setClientInfo(connection, "ApplicationName", "Trino-" + session.getQueryId());
         return connection.prepareCall(procedureQuery.query());
     }
 
+    private void setClientInfo(Connection connection, String key, String value)
+    {
+        try {
+            connection.setClientInfo(key, value);
+        }
+        catch (SQLClientInfoException e) {
+            log.warn("Failed to set client info: %s", e.getMessage());
```

Comment:
Why ignore?

Reply:
> Why ignore?

Because some JDBC drivers may not support the API `java.sql.Connection.setClientInfo()`, I didn't want to fail the query builder just for the client info~
![image](https://github.com/trinodb/trino/assets/26461591/c10f5eb0-3f96-4786-ae68-f4b329a7957a)

Reply:
it shouldn't be a warning either IMO, the admin/user cannot do anything about such a situation anyway.

Reply:
> Because some JDBC drivers may not support the API `java.sql.Connection.setClientInfo()`,

The decision to skip setClientInfo should be done by the connector, not base-jdbc.
And then no need to catch and suppress. And this also avoids non-actionable warning that system admin cannot do anything about.